### ISSUE TITLE
Make the logistic mixture fitting code more robust

### ISSUE THREAD

**Issue by stuhlmueller** [oughtinc/ergo#219]

**Comment by stuhlmueller** [comment]
Demonstrated in https://github.com/oughtinc/ergo/pull/220

**Comment by stuhlmueller** [comment]
Remove `xfail` from tests once this issue is closed.

**Comment by NixGD** [comment]
@stuhlmueller I'm curious for a status update on this.  It's something I've looked into a little bit and would be curious to try to get to the bottom of.  I'm not sure if there's a particular reason the issue was closed.

**Comment by stuhlmueller** [comment]
I assume it works okay now. Do you know if there are tests or demos that show mixture fitting failing?

**Comment by NixGD** [comment]
There are three tests that are set to xfail which all depend on the mixture-fitting. All three of them still fail (one also needed updating to the new interface).

I did a little bit of digging into why.  See [this branch](https://github.com/oughtinc/ergo/tree/demonstrate-fitting-failures) and especially [this notebook](https://github.com/oughtinc/ergo/blob/f30e8a97244ff3eb13f7973d852961aeed5e0de9/notebooks/Fitting%20failure%20demonstration.ipynb).

The short version is I think there are two problems:

1. Ergo implicitly clipping logistic distributions parameters to certain ranges, meaning that it fails to fit weirder distributions.
2. The function that prepares the distribution for metaculus submission assumes the distribution's scale is from 0 to 1, but it isn't always (even if the data is normalized for that range)

**Comment by NixGD** [comment]
6a80b73 should fix point 2 above.
